### PR TITLE
add 2 workflows for DB pooled connections

### PR DESCRIPTION
**Pull request by Aminelouati**
In this PR, we add 2 workflows template SQL_Pooled_Connection_Query and SQL_Pooled_Connection_Update to  demonstrate our ability to use database connections pooling in SQL tasks (basically when running several Tasks concurrently on the same machine).

### CODE REVIEW

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    throw new IllegalArgumentException("You must specify a valid RDBMS name in the script arguments amongst PostgreSQL, MySQL, Greenplum, SQL Server, HSQLDB or Oracle")
+}
+     
+host = variables.get("HOST")
+username = variables.get("USERNAME")
+port = variables.get("PORT")
+database = variables.get("DATABASE")
+
+// This key is used for getting the password from 3rd party credentials.
+CREDENTIALS_KEY = DBMS_PROTOCOL + "://" + username + "@" + host + ":" + port
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: Maybe to put after your test on the port

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            209
+        </positionTop>
+        <positionLeft>
+            370.5
+        </positionLeft>
+      </metadata>
+    </task>
+    <task name="Execute_Query_Statements" >
+      <description>
+        <![CDATA[ This task enables to execute any SQL query statements on an SQL database.
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: This task allow to ..

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+It requires the following third-party credential: {key: rdbms_protocol://<username>@<host>:<port>, value: RDBMS_PASSWORD}. If you connect to your DB without specifying a password, RDBMS_PASSWORD should be a single space string.
+Please refer to the User documentation to learn how to add third-party credentials.
+This task is executed with the following variables:
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: the following variable:

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 4]
```diff
@@ -0,0 +1,306 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="urn:proactive:jobdescriptor:3.11" xsi:schemaLocation="urn:proactive:jobdescriptor:3.11 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.11/schedulerjob.xsd"  name="SQL_Pooled_Connection_Query" projectName="2. SQL" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2" >
+  <variables>
+    <variable name="RDBMS_NAME" value="HSQLDB" model="PA:LIST(PostgreSQL, MySQL, Greenplum, SQL Server, Oracle, HSQLDB)"/>
+  </variables>
+  <description>
+    <![CDATA[ This workflow is a template that demonstrates our ability to use database connections pooling in SQL tasks (basically when running several Tasks concurrently on the same machine). The workflow template is based on the replicate control flow where the split task is Open_Pooled_Connection, the process task is Execute_Query_Statements and finally the merge task is Merge_Result_Set. It is executed using a single variable
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: pooling from SQL tasks

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+  <variables>
+    <variable name="RDBMS_NAME" value="HSQLDB" model="PA:LIST(PostgreSQL, MySQL, Greenplum, SQL Server, Oracle, HSQLDB)"/>
+  </variables>
+  <description>
+    <![CDATA[ This workflow is a template that demonstrates our ability to use database connections pooling in SQL tasks (basically when running several Tasks concurrently on the same machine). The workflow template is based on the replicate control flow where the split task is Open_Pooled_Connection, the process task is Execute_Query_Statements and finally the merge task is Merge_Result_Set. It is executed using a single variable 
+$RDBMS_NAME which enables the user to choose the relational database management system e.g. PostgreSQL, MySQL, Greenplum, HSQLDB, Oracle... 
+
+To allow the persistence of a single connection across multiple tasks executions, the scheduler should be configured in non-fork mode to execute tasks in a single JVM rather than starting a dedicated JVM to run the task (config/scheduler/settings.ini : pa.scheduler.task.fork=false). Besides, you have to download in the addons directory of the scheduler the jdbc connector jar file of the used rdbms. ]]>
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: Besides, you have to download the jdbc connector jar file of the used rdbms, and copy it into the addons directory of the scheduler.

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml [hunk 6]
```diff
@@ -0,0 +1,248 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="urn:proactive:jobdescriptor:3.11" xsi:schemaLocation="urn:proactive:jobdescriptor:3.11 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.11/schedulerjob.xsd"  name="SQL_Pooled_Connection_Update" projectName="2. SQL" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2" >
+  <variables>
+    <variable name="RDBMS_NAME" value="HSQLDB" model="PA:LIST(PostgreSQL, MySQL, Greenplum, SQL Server, Oracle, HSQLDB)"/>
+  </variables>
+  <description>
+    <![CDATA[ This workflow is a template that demonstrates our ability to use database connections pooling in SQL tasks (basically when running several Tasks concurrently on the same machine). The workflow is composed of two tasks Open_Pooled_Connection and Execute_Update_Statements. It is executed using a single variable
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: from SQL tasks

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+  <variables>
+    <variable name="RDBMS_NAME" value="HSQLDB" model="PA:LIST(PostgreSQL, MySQL, Greenplum, SQL Server, Oracle, HSQLDB)"/>
+  </variables>
+  <description>
+    <![CDATA[ This workflow is a template that demonstrates our ability to use database connections pooling in SQL tasks (basically when running several Tasks concurrently on the same machine). The workflow is composed of two tasks Open_Pooled_Connection and Execute_Update_Statements. It is executed using a single variable 
+$RDBMS_NAME which enables the user to choose the relational database management system e.g. PostgreSQL, MySQL, Greenplum, HSQLDB, Oracle... 
+
+To allow the persistence of a single connection across multiple tasks executions, the scheduler should be configured in non-fork mode to execute tasks in a single JVM rather than starting a dedicated JVM to run the task (config/scheduler/settings.ini : pa.scheduler.task.fork=false). Besides, you have to download in the addons directory of the scheduler the jdbc connector jar file of the used rdbms. ]]>
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: Besides, you have to download the jdbc connector jar file of the used rdbms, and copy it into the addons directory of the scheduler.

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+    throw new IllegalArgumentException("You must specify a valid RDBMS name in the script arguments amongst PostgreSQL, MySQL, Greenplum, SQL Server, HSQLDB or Oracle")
+}
+     
+host = variables.get("HOST")
+username = variables.get("USERNAME")
+port = variables.get("PORT")
+database = variables.get("DATABASE")
+
+// This key is used for getting the password from 3rd party credentials.
+CREDENTIALS_KEY = DBMS_PROTOCOL + "://" + username + "@" + host + ":" + port
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: Should be moved after your tests on host and port

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+            370.5
+        </positionLeft>
+      </metadata>
+    </task>
+    <task name="Execute_Update_Statements" >
+      <description>
+        <![CDATA[ This task enables to execute any SQL update statements on an SQL database.
+It requires the following third-party credential: {key: rdbms_protocol://<username>@<host>:<port>, value: RDBMS_PASSWORD}. If you connect to your DB without specifying a password, RDBMS_PASSWORD should be a single space string.
+Please refer to the User documentation to learn how to add third-party credentials.
+This task is executed with the following variables:
```
Comment (mbenguig) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: the following variable:

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml (hunk 3 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: this task is executed with many variables

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml (hunk 1 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: yes u r right

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml (hunk 2 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: ok

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+    init("oracle" ,"1521", "oracle.jdbc.pool.OracleDataSource")
+} else if (DBMS_NAME.equals("HSQLDB")){
+    init("hsqldb:hsql" ,"9001", "org.hsqldb.jdbc.JDBCDataSource")
+} else {
+    throw new IllegalArgumentException("You must specify a valid RDBMS name in the script arguments amongst PostgreSQL, MySQL, Greenplum, SQL Server, HSQLDB or Oracle")
+}
+     
+host = variables.get("HOST")
+username = variables.get("USERNAME")
+port = variables.get("PORT")
```
Comment (fviale) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: If you handle the port with default values, I think the PORT variable should be left blank, and you should assign it to the port local variable if non-empty

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml (hunk 9 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: many variables indeed

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml (hunk 8 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: done

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml (hunk 7 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: done

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml (hunk 6 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: ok

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml (hunk 5 above)
Comment (Aminelouati) [reply DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: ok

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+    init("oracle" ,"1521", "oracle.jdbc.pool.OracleDataSource")
+} else if (DBMS_NAME.equals("HSQLDB")){
+    init("hsqldb:hsql" ,"9001", "org.hsqldb.jdbc.JDBCDataSource")
+} else {
+    throw new IllegalArgumentException("You must specify a valid RDBMS name in the script arguments amongst PostgreSQL, MySQL, Greenplum, SQL Server, HSQLDB or Oracle")
+}
+     
+host = variables.get("HOST")
+username = variables.get("USERNAME")
+port = variables.get("PORT")
```
Comment (fviale) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: same remark as above

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+// This key is used for getting the password from 3rd party credentials.
+CREDENTIALS_KEY = DBMS_PROTOCOL + "://" + username + "@" + host + ":" + port
+CREDENTIALS_KEY_MSG = DBMS_PROTOCOL + "://<username>@<host>:<port>"
+password = credentials.get(CREDENTIALS_KEY)
+
+if (!host) {
+    throw new IllegalArgumentException("ERROR: HOST variable is not provided by the user. Empty value is not allowed.")
+}
+if (!port) {
+    port = DBMS_DEFAULT_PORT
```
Comment (fviale) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: this test is a bit late, as you already used the possibly empty port higher in the code

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+
+// This key is used for getting the password from 3rd party credentials.
+CREDENTIALS_KEY = DBMS_PROTOCOL + "://" + username + "@" + host + ":" + port
+CREDENTIALS_KEY_MSG = DBMS_PROTOCOL + "://<username>@<host>:<port>"
+password = credentials.get(CREDENTIALS_KEY)
+
+if (!host) {
+    throw new IllegalArgumentException("ERROR: HOST variable is not provided by the user. Empty value is not allowed.")
+}
+if (!port) {
```
Comment (fviale) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: same remark as above

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml [hunk 14]
```diff
@@ -0,0 +1,284 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="urn:proactive:jobdescriptor:3.11" xsi:schemaLocation="urn:proactive:jobdescriptor:3.11 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.11/schedulerjob.xsd"  name="SQL_Pooled_Connection_Query" projectName="2. SQL" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2" >
+  <variables>
+    <variable name="RDBMS_NAME" value="HSQLDB" model="PA:LIST(PostgreSQL, MySQL, Greenplum, SQL Server, Oracle, HSQLDB)"/>
+  </variables>
+  <description>
+    <![CDATA[ This workflow is a template that demonstrates our ability to use database connections pooling from SQL tasks (basically when running several Tasks concurrently on the same machine). The workflow template is based on the replicate control flow where the split task is Open_Pooled_Connection, the process task is Execute_Query_Statements and finally the merge task is Merge_Result_Set. It is executed using a single variable
```
Comment (amouhoub) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Query.xml]: This workflow is a template that demonstrates how to use pooled connections to SQL databases across multiple SQL tasks.

File: DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml [hunk 15]
```diff
@@ -0,0 +1,234 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="urn:proactive:jobdescriptor:3.11" xsi:schemaLocation="urn:proactive:jobdescriptor:3.11 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.11/schedulerjob.xsd"  name="SQL_Pooled_Connection_Update" projectName="2. SQL" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2" >
+  <variables>
+    <variable name="RDBMS_NAME" value="HSQLDB" model="PA:LIST(PostgreSQL, MySQL, Greenplum, SQL Server, Oracle, HSQLDB)"/>
+  </variables>
+  <description>
+    <![CDATA[ This workflow is a template that demonstrates our ability to use database connections pooling from SQL tasks (basically when running several Tasks concurrently on the same machine). The workflow is composed of two tasks Open_Pooled_Connection and Execute_Update_Statements. It is executed using a single variable
```
Comment (amouhoub) [inline DataConnectors/resources/catalog/SQL_Pooled_Connection_Update.xml]: Same as above

**Pull request merged**